spigot/src/main/java/de/sean/blockprot/bukkit/integrations/WorldGuardIntegration.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2021 spnda
+ * This file is part of BlockProt <https://github.com/spnda/BlockProt>.
+ *
+ * BlockProt is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BlockProt is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BlockProt.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.sean.blockprot.bukkit.integrations;
+
+import com.sk89q.worldedit.bukkit.BukkitAdapter;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.world.World;
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.protection.ApplicableRegionSet;
+import com.sk89q.worldguard.protection.flags.StateFlag;
+import com.sk89q.worldguard.protection.flags.registry.FlagRegistry;
+import com.sk89q.worldguard.protection.managers.RegionManager;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+import com.sk89q.worldguard.protection.regions.RegionContainer;
+import de.sean.blockprot.bukkit.BlockProt;
+import de.sean.blockprot.bukkit.events.BlockAccessEvent;
+import de.sean.blockprot.bukkit.events.BlockAccessMenuEvent;
+import de.sean.blockprot.bukkit.events.BlockLockOnPlaceEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class WorldGuardIntegration extends PluginIntegration implements Listener {
+    private static final String FLAG_NAME = "allow-blockprot";
+    private static final String CONFIG_ENABLE_FLAG_FUNCTIONALITY = "enable_flag_functionality";
+    private boolean enabled = false;
+
+    private @Nullable Object allowFlag;
```

Comment:
We can use the StateFlag class here again because we catch any NoClassDefErrors when loading anyway.